### PR TITLE
CASSANDRA-17424 Performance and Semantic Concerns w/ Metrics for Local vs. Remote Requests in StorageProxy

### PR DESCRIPTION
**Pull request by maedhroz**

### CODE REVIEW

**Comment by maedhroz** [comment]
https://app.circleci.com/pipelines/github/maedhroz/cassandra?branch=CASSANDRA-17424&filter=all

File: src/java/org/apache/cassandra/service/StorageProxy.java [hunk 1]
```diff
@@ -1027,11 +1011,6 @@ public static void mutateMV(ByteBuffer dataKey, Collection<Mutation> mutations,
                 // add a handler for each mutation - includes checking availability, but doesn't initiate any writes, yet
                 for (Mutation mutation : mutations)
                 {
-                    if (hasLocalMutation(mutation))
-                        writeMetrics.localRequests.mark();
-                    else
-                        writeMetrics.remoteRequests.mark();
-
```
Comment (maedhroz) [inline src/java/org/apache/cassandra/service/StorageProxy.java:1034]: This codepath seems to be actuated by writes on the MV's base table, so it didn't seem correct to record anything around these metrics.

File: src/java/org/apache/cassandra/service/StorageProxy.java [hunk 2]
```diff
@@ -1684,6 +1670,9 @@ protected Verb verb()
             // we build this ONLY to perform the sufficiency check that happens on construction
             ReplicaPlans.forWrite(keyspace, cm.consistency(), tk, ReplicaPlans.writeAll);
 
+            // This host isn't a replica, so mark the request as being remote.
+            writeMetrics.remoteRequests.mark();
+
```
Comment (maedhroz) [inline src/java/org/apache/cassandra/service/StorageProxy.java:1677]: If this host is a replica, `applyCounterMutationOnCoordinator()` -> `performWrite()`, and there we'll mark a local request against the metrics.

Comment (jonmeredith) [reply src/java/org/apache/cassandra/service/StorageProxy.java:1677]: Is it worth being explicit in this comment that the other path will ultimately update the `localRequests`

**Comment by jonmeredith** [comment]
Looks like the original test is missing anything to test the counter path.

File: src/java/org/apache/cassandra/service/reads/AbstractReadExecutor.java [hunk 3]
```diff
@@ -17,10 +17,8 @@
  */
 package org.apache.cassandra.service.reads;
 
-import java.util.List;
-import java.util.stream.Collectors;
-
 import com.google.common.base.Preconditions;
+import org.apache.cassandra.utils.FBUtilities;
```
Comment (jonmeredith) [inline src/java/org/apache/cassandra/service/reads/AbstractReadExecutor.java]: Out of order import

**Review by jonmeredith — APPROVED** [review]

**Review by jonmeredith — APPROVED** [review]

**Comment by maedhroz** [comment]
Committed as https://github.com/apache/cassandra/commit/57ab3afcf16970047d3df4656241cf0705e94bee

**Pull request closed without merging**